webapp/components/test-runs-query-builder.html

```diff
@@ -383,6 +401,52 @@ <h3>
           labels: (product.labels || []).filter(l => !CHANNELS.has(l)).concat(channel)
         });
       }
+
```

Comment:
It's hard to keep the flow straight my method name alone in this block of methods. Could we add short header comments like `// Respond to [foo] changing by [action]`, where `[foo]` and `[action]` are more than just a property or URL name. E.g., "Respond to list of versions to be suggested in data list changing by ..."

Reply:
Done.